### ISSUE TITLE
Intermittent freezes

### ISSUE THREAD

**Issue by thany** [transmission-remote-gui/transgui#1298]
Any action that involves communication with the server may or may not cause the *whole* application to freeze for a moment or two. It may or may not go into "Not responding" mode. And it may or may not mess up its GUI while frozen. It also may or may not turn up a "Bad Request" error. Not all freezes result in this error, but this error *always* comes after a freeze - never on its own.

I understand that sometimes the application needs to wait for the server to respond, but for god's sake do it in the background, please. Or asynchronouly at least.

A side-effect of these freezes is that it may sometimes "miss" a torrent getting added, or some other action is missed. After it unfreezes, say, 20% or so of the time, the torrent/magnet I clicked is *not* added, regardless of whether I got that "Bad Request" error.

Another side-effect is that (sporadically though) a secondary instance of the program is launched, which is only adding more confusion. Which one is now the one catching clicks on magnets? One of the two, and it's always unluckily the one I decide close.

TL;DR
Please wait for server responses in the background or asynchronously. And keep a queue of actions to perform on the server, so that requested actions aren't lost.

Versions:
* Windows 10
* Transmission server 2.94
* Transmission GUI 5.18.0

**Comment by username-not-taken** [comment]
i think that it happens because it jumps back to the all torrents tab when you add more torrents and with hundreds or even thousands torrents getting and rendering the stats for all of them is a very heavy operation. this thing really needs a option that allows setting the default tab so that it never jumps back to the all torrents tab

**Comment by leonsoft-kras** [comment]
The program is written in Pascal and its maintenance is not an easy task. 
Your suggestion is clear. However, it requires significant reworking of the program. Offer your help.

I found a simple way (as probably many).
1. transmission shutdown (loading, uploading)
2. adding all new torrents
3. start up the transmission (now max speed - loading, uploading)

works great.

**Comment by leonsoft-kras** [comment]
#1304

**Comment by username-not-taken** [comment]
never seen any pascal code before but i guess that i could hardcode the things that i want. should not be too hard to figure out how to do it since something already defines the defaults

**Comment by rezad1393** [comment]
I have the same issue but on archlinux.

**Comment by thany** [comment]
@leonsoft-kras 
> The program is written in Pascal and its maintenance is not an easy task.

No offense, but that tells me some refactoring is due, regardless of this issue.

**Comment by leonsoft-kras** [comment]
> 
> 
> @leonsoft-kras
> 
> > The program is written in Pascal and its maintenance is not an easy task.
> 
> No offense, but that tells me some refactoring is due, regardless of this issue.

Good luck. We dont mind. There are only 100,000 lines. Change the code. Check for changes.
Advising is always easier than making an effort.

P.S. The code is not mine, so I will not take up this work, neither free nor paid.
